src/core/net/ip6.cpp

```diff
@@ -1058,6 +1058,12 @@ const NetifUnicastAddress *Ip6::SelectSourceAddress(MessageInfo &aMessageInfo)
             rvalAddr          = addr;
             rvalPrefixMatched = candidatePrefixMatched;
         }
+
```

Comment:
is it better to add here:
```cpp
else
{
    continue;
}
```

This avoids recalculate `destinationScope` when `rvalAddr` is not changed.